### ISSUE TITLE
Element/Component runtime refactor

### ISSUE THREAD

**Issue by webberwang** [codelab-app/platform#3105]
## Description

- Move properties on `Element` and `Component` that deals with runtime props
- Refactor `RuntimeElementProp` and `RuntimeComponentProp`, to `RuntimeElement` and `RuntimeComponent`
- We treat the runtime models as the node, the ref allows us extend the original model and add runtime info on there. https://refactoring.guru/design-patterns/decorator 
- I tried having `Renderer.runtimeElements: Array<IRuntimeComponent | IRuntimeElement>`, but ended up doing `Renderer.runtimeComponents`, `Renderer.runtimeElements` as they seem different enough, and we don't have to constantly check for type

```typescript
interface IRuntimeElement {
  runtimeStore: IRuntimeStore
  
  // Props directly on element
  // expressionEvaluationContext etc
  *: IPropData
}

interface IRuntimeStore {
  runtimeActions: Array<IRuntimeAction>
  // State & Refs can go here
}
```

- Basically, `Renderer.addRuntimeElement()` in the `Renderer.renderIntermediateElement` is what start the transformation process of converting `IElementModel` to `IRuntimeElement`
- Can think of `IRuntimeElement` as a place to put computed properties in the application layer. Everything can be computed from `IElementModel` and its relationships
- Originally thought of having a `propsPipe` on `RuntimeElement`, but having computed on `RuntimeElement` seems sufficient. Could have performance impact if each element has 5 pipes (seems unnecessary). The goal was to pull runtime props out of ElementModel
- Advantage of this new approach is we wouldn't need to clone components, but have a better way to represent the model
---

**Comment by assimhabeek** [comment]
I can start by these two : 
- [ ] Remove cloning logic (element/component/store...)
- [ ] Add specs for runtime models. (we need be certain that data flow is correct)
